js.go

```diff
@@ -165,10 +165,27 @@ func (opt jsOptFn) configureJSContext(opts *js) error {
 // APIPrefix changes the default prefix used for the JetStream API.
 func APIPrefix(pre string) JSOpt {
 	return jsOptFn(func(js *js) error {
-		js.pre = pre
-		if !strings.HasSuffix(js.pre, ".") {
-			js.pre = js.pre + "."
+		pre = strings.TrimSuffix(pre, ".")
+
+		if pre == ">" {
+			return ErrJetStreamBadPre
 		}
+
+		toks := strings.Split(pre, ".")
+		for i, tok := range toks {
+			if tok == "" || tok == "*" {
+				return ErrJetStreamBadPre
+			}
+
+			if tok == ">" && i < len(toks)-1 {
+				return ErrBadSubject
+			} else if tok == ">" && i == len(toks)-1 {
+				js.pre = pre
```

Comment:
I am confused. If it is a prefix, why are we allowing something like `foo.>` here? being a prefix it means that it would then possibly added to something and then become `foo.>.bar`? But maybe I don't understand what that is.